packages/agents/src/agents/Chameleon/index.ts

```diff
@@ -183,18 +120,15 @@ export class ChameleonAgent extends NewAgent<GoalRunArgs> {
   };
 
   private async shortenLargeMessages(query: string): Promise<void> {
-    for(let i = 2; i < this.context.chat.chatLogs.messages.length ; i++) {
+    for(let i = 0; i < this.context.chat.chatLogs.messages.length ; i++) {
       const message = this.context.chat.chatLogs.messages[i];
-      if (this.isLargeMsg(message)) {
-        await this.shortenMessage(message, query);
+      if (i <= 3 || !this.isLargeMsg(message)) {
+      } else {
+        message.content = await this.advancedFilterText(message.content ?? "", query);
```

Comment:
Can't we invert the condition and have no `else` and no empty `if` :D ? Like: `i > 3 && this.isLargeMsg(message)`